src/Workspaces/Remote/ServiceHub/Services/DiagnosticAnalyzer/DiagnosticComputer.cs

```diff
@@ -90,25 +134,8 @@ internal class DiagnosticComputer
 
             var skippedAnalyzersInfo = _project.GetSkippedAnalyzersInfo(_analyzerInfoCache);
 
-            try
-            {
-                return await AnalyzeAsync(compilationWithAnalyzers, analyzerToIdMap, analyzers, skippedAnalyzersInfo,
-                    reportSuppressedDiagnostics, logPerformanceInfo, getTelemetryInfo, cancellationToken).ConfigureAwait(false);
-            }
-            catch
-            {
-                // Do not re-use cached CompilationWithAnalyzers instance in presence of an exception, as the underlying analysis state might be corrupt.
```

Comment:
This logic is no longer required as `CompilationWithAnalyzers` doesn't do partial analysis state tracking anymore.

Reply:
👍

Reply:
So, this identified a bug in the core AnalyzerDriver in presence of cancellations. The cancellationToken passed in is being captured in a lambda, and this same token is re-used when another call is made into a driver with a different non-cancelled token. I am going to revert this change to remove the try/catch and open a separate PR to address this bug.

Reply:
#67084